crates/bevy_ecs/macros/src/component.rs

```diff
@@ -92,11 +92,16 @@ pub fn derive_component(input: TokenStream) -> TokenStream {
             });
             match &require.func {
                 Some(RequireFunc::Path(func)) => {
+                    let constructor = if func.is_ident("explicit") {
+                        quote! { None }
+                    } else {
+                        quote! {Some(|| { let x: #ident = #func().into(); x })}
+                    };
```

Comment:
This could be surprising behavior to users who name their constructor `explicit` (for whatever reason). Could we make sure this is at least documented?

Alternatively we could do `#[requires(explicit = true)]` and peeking for `=` could be used to disambiguate the two. Not sure that's totally worth the ergonomics hit though.